### ISSUE TITLE
Support for reading and writing streaming frames websockets

### ISSUE THREAD

**Issue by shaohme** [fafhrd91/aiohttp#7787]
### Is your feature request related to a problem?

I sometimes need to read/write a large incoming binary websocket stream of frames before its finished (FIN) frame arrives, so that it does not need to buffer everything before handing my application an assembled binary websocket message. I accept, of course, that the transport could be interrupted at any time.

### Describe the solution you'd like

aiohttp already implements [https://docs.aiohttp.org/en/stable/streams.html](streaming) and a similar API could be nice to have for its websocket-request/response implementation.

### Describe alternatives you've considered

python-websockets [https://websockets.readthedocs.io/en/stable/](url) already have this, but only limited to its Sans writer, and in its 'threading/sync' API.

### Related component

Server, Client

### Code of Conduct

- [X] I agree to follow the aio-libs Code of Conduct

**Comment by Dreamsorcerer** [comment]
You mean instead of getting a complete message with .receive(), you'd be able to stream the msg.data as it arrives (presumably the msg.type would be available immediately though, similar to resp.headers)?

I think that's a reasonable request, although I'm not sure what the API would look like, if someone wants to volunteer to work on it.

**Comment by shaohme** [comment]
> You mean instead of getting a complete message with .receive(), you'd be able to stream the msg.data as it arrives (presumably the msg.type would be available immediately though, similar to resp.headers)?

Yes. I intend to transfer large payloads over websockets, and I don't want to buffer it all at once before start sending the chunks off elsewhere.

> 
> I think that's a reasonable request (although I'm not sure what the API would look like), if someone wants to volunteer to work on it.
